src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,18 +209,18 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                     //
                     // Avoid instance state modification
-                    Dictionary<string, ConfigurationSetting> applicationSettings = null;
                     Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-                    Dictionary<KeyValueIdentifier, KeyValueChange> keyValueChanges = null;
+                    List<KeyValueChange> keyValueChanges = null;
                     List<KeyValueChange> changedKeyValuesCollection = null;
+                    Dictionary<string, ConfigurationSetting> data = null;
                     bool refreshAll = false;
                     StringBuilder logInfoBuilder = new StringBuilder();
                     StringBuilder logDebugBuilder = new StringBuilder();
 
                     await ExecuteWithFailOverPolicyAsync(availableClients, async (client) =>
                         {
-                            applicationSettings = null;
-                            keyValueChanges = new Dictionary<KeyValueIdentifier, KeyValueChange>();
+                            data = null;

```

Comment:
Should I also be setting watchedSettings to null here following the same idea as data?

Reply:
Looks like with current usage the answer is no since it's only set right before exiting `ExecuteWithFailoverPolicyAsync`. But I wouldn't argue against adding it for symmetry purposes and just saying that the variables that are declared outside should be reset at the beginning of `ExecuteWithFailoverPolicyAsync`.